_episodes/10-distribution.md

```diff
@@ -0,0 +1,245 @@
+environment.
+What does this mean?
+More importantly, what does it have to do with letting other people do
+`pip install molecool` and get my package automatically downloaded from
+"the cloud"?
+
+Setuptools is not the only package for configuring and building Python
+distribution archives,
+but it is what we will tell `pip` to use for this project.
+When `pip install` is run on package sources, it notices the `pyproject.toml` file (which marks a folder as containing
+buildable packages),
+launches the configured build system, and then installs the resulting "wheel" (distribution archive).
+Note that, if we _only_ want to prepare a distribution for sharing,
+we can just use the `build` module.
+
+~~~
+pip install build
+python -m build .
+~~~
+{: .language-bash}
+
+
+Note the version string embedded in the distribution archive name.
+`versioningit` (configured in `pyproject.toml`) also participated in the build process.
+`versioningit` generated a `_version.py` file with a version string derived
+from your most recent `git` tag. We will not add this file to our `git`
+revision control, but it will be packaged in our distribution.
+`versioningit` will regenerate this file as necessary and appropriate.
+
+Note that the untagged post-release version strings
+(using our formatting rules in `pyproject.toml`)
+do not satisfy PEP 440 and will be rejected if we try to upload to pypi.
+This is probably a good thing; it will help keep us from accidentally publishing
+releases before they are ready, or messing up our versioning scheme by forgetting
+a semantic versioning tag.
+
+> ## Goodbye to setup.py?
+>
+> Historically packages built with `setuptools` used a `setup.py` script
+as the entry point to the `setuptools` build system. `setup.py` is specific
+to `setuptools` and the `distutils` package it is built on.
+Many tools for helping to build and test packages came to
+rely on `setup.py`, which made them incompatible with non-setuptools build
+systems.
+> 
+> In an ongoing effort at generalization, the Python community is now
+encouraging package authors to use a `pyproject.toml` file to direct the build
+system. `distutils` is scheduled for retirement with Python 3.12.
+In preparation, `setuptools` no longer requires `distutils` or `setup.py`,
+and can be configured entirely with `pyproject.toml`.
+{: .callout}
+
+### Building and uploading to PyPI
+
+Refer to https://packaging.python.org/en/latest/tutorials/packaging-projects/
+for build-system-agnostic tutorial on
+* packaging Python projects for distribution
+* publishing packages to pypi.org for automatic download with `pip install`
+
+You will note that most of the configuration described in the packaging tutorial
+has already been pre-configured by the cookiecutter.
+
+Essentially, once you have an account on `pypi.org` and you have built
+a distribution archive with `python -m build`, you can `pip install twine` and
+use twine to upload your package, per the above linked tutorial.
```

Comment:
```suggestion
Read through https://packaging.python.org/en/latest/tutorials/packaging-projects/
for a build-system-agnostic tutorial on
* packaging Python projects for distribution
* publishing packages to pypi.org for automatic download with `pip install`

By the end of the [tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/),
you will understand how to get an account on `pypi.org`,
build a distribution archive with `python -m build`,
and use twine to upload your package.

The [tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/) should go quickly because,
as you will note, most of the configuration it describes
has already been pre-configured by the cookiecutter.
```